homeassistant/components/mysensors.py

```diff
@@ -286,8 +300,14 @@ def mysensors_callback(gateway, msg):
                 else:
                     devices[key].update()
                 continue
+            node_name = msg.node_id
+            for _, node in enumerate(gateway.nodes_config):
+                if (node.get(CONF_NODE_NAME) is not None and
+                        node.get(CONF_NODE_ID) is msg.node_id):
+                    node_name = node.get(CONF_NODE_NAME)
+                    break
             name = '{} {} {}'.format(
-                gateway.sensors[msg.node_id].sketch_name, msg.node_id,
+                gateway.sensors[msg.node_id].sketch_name, node_name,
```

Comment:
I think the whole name should be replaced, not just the node id. If the user wants to change name, the user takes responsibility for the whole name.